org.eclipse.scanning.event/src/org/eclipse/scanning/event/ConsumerImpl.java

```diff
@@ -488,121 +475,107 @@ private void init() throws EventException {
-			logger.error("Fatal except deserializing object!", ne);
-			return true;
-		}
-		if (ne.getClass().getSimpleName().endsWith("UnrecognizedPropertyException")) {
-			logger.error("Cannot deserialize bean!", ne);
+		// if error occurred deserializing the bean, log a specific message and don't fail even if not durable
+		if (e.getClass().getSimpleName().contains("Json") || e.getClass().getSimpleName().endsWith("UnrecognizedPropertyException")) {
+			LOGGER.error("Could not deserialize bean.", e);
 			return true;
 		}
 
-		if (ne.getClass().getSimpleName().endsWith("ClassCastException")) {
-			logger.error("Problem with serialization?", ne);
+		// normal error case, log error message and continue if
+		if (e instanceof EventException || e instanceof InterruptedException) {
+			if (Thread.interrupted()) return false;
+			LOGGER.error("Cannot consume message ", e);
+			return isDurable();
 		}
 
-
 		if (!isDurable()) return false;
 
+		// TODO: below assumes some conenction problem with activemq. why would any remaining
+		// exception be a connection problem?
+		LOGGER.warn("{} ActiveMQ connection to {} lost.", getName(), uri, e);
+		LOGGER.warn("We will check every 2 seconds for 24 hours, until it comes back.");
 		try {
 			if (Thread.interrupted()) return false;
+			// Wait for 2 seconds (default time)
 			Thread.sleep(Constants.getNotificationFrequency());
-		} catch (InterruptedException e) {
-			throw new EventException("The consumer was unable to wait!", e);
+		} catch (InterruptedException ie) {
+			throw new EventException("The consumer was unable to wait!", ie);
 		}
 
-		waitTime+=Constants.getNotificationFrequency();
-		checkTime(waitTime);
-
-		logger.warn(getName()+" ActiveMQ connection to "+uri+" lost.");
-		logger.warn("We will check every 2 seconds for 24 hours, until it comes back.");
+		waitTime += Constants.getNotificationFrequency();
+		checkTime(waitTime); // Exits if wait time more than one day
 
 		return true;
 	}
 
 	private void checkStartPaused() throws EventException {
-
 		if (!isPauseOnStart()) {
 			return;
 		}
 
 		List<U> items = getSubmissionQueue();
-		if (items!=null && items.size()>0) {
+		if (items != null && !items.isEmpty()) {
 			pause();
 
 			IPublisher<PauseBean> pauser = eservice.createPublisher(getUri(), IEventService.CMD_TOPIC);
-			pauser.setStatusSetName(IEventService.CMD_SET); // The set that other clients may check
+			pauser.setStatusSetName(EventConstants.CMD_SET); // The set that other clients may check
 			pauser.setStatusSetAddRequired(true);
 
 			PauseBean pbean = new PauseBean();
 			pbean.setQueueName(getSubmitQueueName()); // The queue we are pausing
 			pbean.setPause(true);
 			pauser.broadcast(pbean);
-
 		}
 	}
 
 	private void checkPaused() throws Exception {
 
-		if (!isActive()) throw new Exception("The consumer is not active and cannot be paused!");
+		if (!isActive())
+			throw new Exception("The consumer is not active and cannot be paused!");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)